src/main/scala/com/datastax/sparkstress/DataGenerator.scala

```diff
@@ -97,41 +141,49 @@ object RowGenerator {
   val sizes = List("P", "S", "M", "L", "XL", "XXL", "XXXL").view
   val qtys = (5 to 10000 by 5).view
   val perftime = new DateTime(2000,1,1,0,0,0,0)
-  val perfRandom = 42
 
   class PerfRowGenerator(numPartitions: Int, numTotalRows: Long, numTotalKeys: Long)
     extends RowGenerator[PerfRowClass]() {
 
     val clusteringKeysPerPartitionKey = numTotalRows / numTotalKeys
     val partitionKeysPerSparkPartition = numTotalKeys / numPartitions
 
-    override def generatePartition(index: Int): Iterator[PerfRowClass] = {
+    override def generatePartition(seed: Long, index: Int): Iterator[PerfRowClass] = {
       val offset = partitionKeysPerSparkPartition * index;
-      val r = new scala.util.Random(index * perfRandom)
+      val r = new scala.util.Random(index * seed)
 
       for ( pk <- (1L to partitionKeysPerSparkPartition).iterator; ck <- (1L to clusteringKeysPerPartitionKey).iterator) yield {
         val color = colors(r.nextInt(colors.size))
         val size = sizes(r.nextInt(sizes.size))
         val qty = qtys(r.nextInt(qtys.size))
         val store = s"Store ${pk + offset}"
-        val order_number = UUID.randomUUID()
-        val order_time = perftime.plusSeconds(r.nextInt(1000))
+        val order_number = UUID.randomUUID().toString()
+        val order_time = perftime.plusSeconds(r.nextInt(1000)).toString()
         PerfRowClass(store, order_time, order_number, color, size, qty)
       }
     }
   }
 
-  def getPerfRowRdd(sc: SparkContext, numPartitions: Int, numTotalRows: Long, numTotalKeys: Long): RDD[PerfRowClass] = {
-
+  def getPerfRowRdd(ss: SparkSession, seed: Long, numPartitions: Int, numTotalRows: Long, numTotalKeys: Long): RDD[PerfRowClass] = {
     val perfRowGenerator = new PerfRowGenerator(numPartitions, numTotalRows, numTotalKeys)
-
-    sc.parallelize(Seq[Int](), numPartitions).mapPartitionsWithIndex {
+    ss.sparkContext.parallelize(Seq[Int](), numPartitions).mapPartitionsWithIndex {
       case (index, n) => {
-        perfRowGenerator.generatePartition(index)
+        perfRowGenerator.generatePartition(seed, index)
       }
     }
   }
 
+  def getPerfRowDataset(ss: SparkSession, seed: Long, numPartitions: Int, numTotalRows: Long, numTotalKeys: Long): org.apache.spark.sql.Dataset[PerfRowClass] = {
```

Comment:
why isn't this just "getPerfRowRDD.toDS"?

Reply:
Yeah basically all of these can just call their sub methods and "toDS" themselves. Should save a bit of code

Reply:
I think I addressed all these cases.